src/GenCtypes.ml

```diff
@@ -0,0 +1,276 @@
+    let p = Pat.mk (Ppat_var (mk_sym unqual_name)) in
+    mk_decl ?t:(Some t) p e
+  in
+  let rec mk_field anon (f_name, f_typ) =
+    match f_name with
+    | Some name ->
+      let p = Pat.mk (Ppat_var (mk_sym (unqual_name ^ "_" ^ name))) in
+      let c_name = if anon then "" else name in
+      let e = mk_app (exp_ident "field") [exp_ident unqual_name; mk_const c_name; mk_typ module_name f_typ] in
+      [[Vb.mk p e] |> Str.value Nonrecursive]
+    | None -> (* an anonymous union in a struct *)
+      begin match f_typ with
+      | Union fields ->
+          let name = unqual_name ^ "__u" in
+          let fields = List.map (fun (x, y) -> (Some x, y)) fields in
+          (mk_struct_decl Union module_name name fields) @ (mk_field true (Some "u", Qualified name))
+      | _ -> Warn.fatal_error "Unreachable"
+      end
+  in
+  let type_decl = Str.type_ Recursive [Type.mk ?manifest:(Some tm) (mk_sym unqual_name)] in
+  let seal_decl = mk_decl (Pat.any ()) (mk_app (exp_ident "seal") [exp_ident unqual_name]) in
+  [type_decl; ctypes_structure] @ (KList.map_flatten (mk_field false) fields) @ [seal_decl]
+
+let mk_typedef module_name name typ =
+  let typ_name = mk_unqual_name module_name name in
+  [ Str.type_ Recursive [Type.mk ?manifest:(Some uint8_t) (mk_sym typ_name)]
+  ; mk_simple_app_decl typ_name None "typedef" [mk_typ module_name typ; mk_const name] ]
+
+let mk_enum_tags module_name name tags =
+  let rec mk_tags n tags =
+    match tags with
+    | [] -> []
+    | t :: ts ->
+      let tag_name = String.concat "_" (List.map (mk_unqual_name module_name) [name; t]) in
+      (mk_simple_app_decl tag_name None "Unsigned.UInt8.of_int"
+         [Exp.constant (Const.int n)]) :: (mk_tags (n+1) ts)
+  in
+  mk_tags 0 tags
+
+let build_foreign_exp module_name name return_type parameters : expression =
+  let types = List.map (fun n -> mk_typ module_name n.typ) parameters in
+  let returning = mk_app (exp_ident "returning") [mk_typ module_name return_type] in
+  let e = List.fold_right (fun t1 t2 -> mk_app (exp_ident "@->") [t1; t2]) types returning in
+  mk_app (exp_ident "foreign") [mk_const name; e]
+
+let build_binding module_name name return_type parameters : structure_item =
+  let func_name = KString.chop name (module_name ^ "_") in
+  let e = build_foreign_exp module_name name return_type parameters in
+  let p = Pat.mk (Ppat_var (mk_sym func_name)) in
+  mk_decl p e
+
+let mk_ctypes_decl module_name (d: decl): structure =
+  match d with
+  | Function (_, _, return_type, name, parameters, _) ->
+      (* Don't generated bindings for projectors and internal names *)
+      if not (KString.starts_with name (module_name ^ "___proj__")) &&
+         not (KString.starts_with name (module_name ^ "_uu___"))then
+        [build_binding module_name name return_type parameters]
+      else
+        []
+  | Type (name, typ, _) -> begin (* VD: do I need to consider flags here? *)
```

Comment:
No you can safely ignore them. The only one you might possibly be interested in is `Const` but it's a poorly designed feature and no one uses it anyhow